my_enums.rb

```diff
@@ -1,60 +1,138 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
-
 module Enumerable #:nodoc: all
   def my_each
+    return to_enum unless block_given?
+
     length.times do |n|
       current = self[n]
       yield(current)
     end
+    self
   end
 
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return to_enum unless block_given?
```

Comment:
* [ ] ``` my_each_with_index ``` Must return enumerator when no block is given. the code below return nothing
``` array = [1,2,5,4,7]
array.my_each_with_index```